plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -1883,6 +1924,410 @@ public void testQueryTableFunction()
+    {
+        String tableName = "test_highly_nested_data_" + randomNameSuffix();
+
+        try {
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 1)
+                    .put("row1_t", ImmutableMap.<String, Object>builder()
+                            .put("f1", 2)
+                            .put("f2", 3)
+                            .put("row2_t", ImmutableMap.<String, Object>builder()
+                                    .put("f1", 4)
+                                    .put("f2", 5)
+                                    .put("row3_t", ImmutableMap.<String, Object>builder()
+                                            .put("f1", 6)
+                                            .put("f2", 7)
+                                            .buildOrThrow())
+                                    .buildOrThrow())
+                            .buildOrThrow())
+                    .buildOrThrow());
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 11)
+                    .put("row1_t", ImmutableMap.<String, Object>builder()
+                            .put("f1", 12)
+                            .put("f2", 13)
+                            .put("row2_t", ImmutableMap.<String, Object>builder()
+                                    .put("f1", 14)
+                                    .put("f2", 15)
+                                    .put("row3_t", ImmutableMap.<String, Object>builder()
+                                            .put("f1", 16)
+                                            .put("f2", 17)
+                                            .buildOrThrow())
+                                    .buildOrThrow())
+                            .buildOrThrow())
+                    .buildOrThrow());
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 21)
+                    .put("row1_t", ImmutableMap.<String, Object>builder()
+                            .put("f1", 22)
+                            .put("f2", 23)
+                            .put("row2_t", ImmutableMap.<String, Object>builder()
+                                    .put("f1", 24)
+                                    .put("f2", 25)
+                                    .put("row3_t", ImmutableMap.<String, Object>builder()
+                                            .put("f1", 26)
+                                            .put("f2", 27)
+                                            .buildOrThrow())
+                                    .buildOrThrow())
+                            .buildOrThrow())
+                    .buildOrThrow());
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Test select projected columns, with and without their parent column
+        assertQuery("SELECT id, row1_t.row2_t.row3_t.f2 FROM " + tableName, "VALUES (1, 7), (11, 17), (21, 27)");
```

Comment:
Use `assertThat(query(...))` here and everywhere else, instead of `assertQuery`.